pkg/registry/event/strategy.go

```diff
@@ -71,31 +70,31 @@ func (eventStrategy) AllowUnconditionalUpdate() bool {
 }
 
 func MatchEvent(label labels.Selector, field fields.Selector) *generic.SelectionPredicate {
-	return &generic.SelectionPredicate{Label: label, Field: field, GetAttrs: getAttrs}
-}
-
-func getAttrs(obj runtime.Object) (objLabels labels.Set, objFields fields.Set, err error) {
-	event, ok := obj.(*api.Event)
-	if !ok {
-		return nil, nil, errors.NewInternalError(fmt.Errorf("object is not of type event: %#v", obj))
-	}
-	l := event.Labels
-	if l == nil {
-		l = labels.Set{}
+	return &generic.SelectionPredicate{
+		Label: label,
+		Field: field,
+		GetAttrs: func(obj runtime.Object) (labels.Set, fields.Set, error) {
+			event, ok := obj.(*api.Event)
+			if !ok {
+				return nil, nil, fmt.Errorf("not an event")
+			}
+			return labels.Set(event.Labels), SelectableFields(event), nil
+		},
 	}
+}
 
-	objectMetaFieldsSet := generic.ObjectMetaFieldsSet(event.ObjectMeta, true)
-	specificFieldsSet := fields.Set{
-		"involvedObject.kind":            event.InvolvedObject.Kind,
-		"involvedObject.namespace":       event.InvolvedObject.Namespace,
-		"involvedObject.name":            event.InvolvedObject.Name,
-		"involvedObject.uid":             string(event.InvolvedObject.UID),
-		"involvedObject.apiVersion":      event.InvolvedObject.APIVersion,
-		"involvedObject.resourceVersion": event.InvolvedObject.ResourceVersion,
-		"involvedObject.fieldPath":       event.InvolvedObject.FieldPath,
-		"reason":                         event.Reason,
-		"source":                         event.Source.Component,
-		"type":                           event.Type,
-	}
-	return l, generic.MergeFieldsSets(objectMetaFieldsSet, specificFieldsSet), nil
+func SelectableFields(event *api.Event) fields.Set {
+	fields := make(fields.Set, 12)
```

Comment:
Perhaps it'd be better to just explicitly add the elements from that here, and then have a test that it includes everything that AddObjectMetaFields would have added.